gateway-sp-comms/src/single_sp.rs

```diff
@@ -740,10 +751,66 @@ impl SingleSp {
     /// Instruct the SP to reset a component.
     ///
     /// Only valid after a successful call to `reset_component_prepare()`.
+    ///
+    /// If `disable_watchdog` is `true`, then any watchdogs associated with the
+    /// reset are disabled.  Otherwise, watchdogs are enabled opportunistically
+    /// (depending on component and MGS protocol version).
     pub async fn reset_component_trigger(
         &self,
         component: SpComponent,
+        disable_watchdog: bool,
     ) -> Result<()> {
+        // If the SP has an update pending, then use the watchdog reset
+        let mut use_watchdog = component == SpComponent::SP_ITSELF
```

Comment:
Now that we're taking a `component` in the message, should we drop this check for `SP_ITSELF` here? Relatedly, do you think `component_watchdog_supported` should return a `bool` instead of just an ack? I think if it did, that would make "is the watchdog supported for this component" is an SP-level decision instead of an MGS-level decision.

Reply:
(I guess the "clearing a watchdog" logic would similarly have to change, although maybe that is also just removing the `SP_ITSELF` check and relying on `use_watchdog`?)

Reply:
I think this is reasonable; done in 78b7356edbea64fe30e96b1451806037f4a80f71 (along with some other cleanup)